core/cap-0023.md

```diff
@@ -239,146 +293,248 @@ default:
+issuer does want to pre-authorize all of those accounts, they cannot use this
+feature. But this is most useful for asset issuance, in which case there is
+likely to be only one account. In any case, if the issuer is using this
+functionality then they must be cognizant about clearing the `authorize` flag
+for any `ClaimableBalanceEntry` claimable by an account from which they want to
+revoke authorization.
+
+### Should it be possible to increase the amount of a ClaimableBalanceEntry?
+One issue which has been discussed during the development of this proposal is
+the absence of a mechanism to increase the `amount` of a
+`ClaimableBalanceEntry`. The specific scenario which would warrant this
+functionality is when a single account sends many identical payments to a single
+account that is not prepared to receive them and does not claim any of the
+payments. However, this case is sufficiently specific that we recommend pursuing
+it in a separate proposal once this proposal has been implemented. Delaying this
+feature presents minimal additional difficulty because `ClaimableBalanceEntry`
+has an extension point.
+
+### Memo
+Everything proposed in this document takes the same stance as existing features
+of the protocol with regard to memo: memo is a property of a transaction, not of
+an operation or a ledger entry.
+
+### Example: Revocable Payment
+An earlier draft of this proposal focused on revocable payments. A revocable
+payment temporarily grants the option of permanent ownership to a specified
+party. For example, a revocable payment of `100 X` from `A` to `B` on December
+9th, 2019 might grant to `B` until December 16th, 2019 the option of permanent
+ownership. Claiming permanent ownership costs only a transaction fee, so why
+might `B` choose not to claim? One answer is that `B` may want to claim, but is
+unable to because they have lost access to their account. Revocable payments
+give senders the opportunity to retry a payment if the recipient is unable to
+receive it.
+
+We now consider the general case, in which the owner of account `A` wants to
+send a revocable payment of `<AMOUNT>` of asset `X` to account `B` expiring at
+time `T`. This can be done with the following `Operation`:
+```
+sourceAccount: A
+body:
+    type: CREATE_CLAIMABLE_BALANCE
+    createClaimableBalanceOp:
+        asset: X
+        amount: <AMOUNT>
+        condition[0]:
+            destination: B
+            weight: 1
+            predicate:
+                type: CLAIM_PREDICATE_BEFORE_ABSOLUTE_TIME
+                absBefore: T
+        condition[1]:
+            destination: A
+            weight: 1
+            predicate:
+                type: CLAIM_PREDICATE_AFTER_ABSOLUTE_TIME
+                absAfter: T
+        threshold: 1
+```
 
 ## Backwards Incompatibilities
 All downstream systems will need updated XDR in order to recognize the new
 operations and ledger entries.
 
 ## Security Concerns
```

Comment:
One use case this claimable capability could open up, if it didn't include the reserve, is the ability for payment products that send payments to give recipients the opportunity to decline the receipt of a payment. This isn't part of the motivation at the moment by the looks of it, but this operation would support that use case if the reserve was optional or was sent back.